swo/mpt/client/base.py

```diff
@@ -14,12 +15,34 @@ def __init__(self, base_url, api_token):
         )
         self.mount("http://", HTTPAdapter(max_retries=retries))
         self.headers.update(
-            {"Authorization": f"Bearer {api_token}", "User-Agent": "swo-extensions/1.0"}
+            {"User-Agent": "swo-extensions/1.0"}
         )
         self.base_url = f"{base_url}/" if base_url[-1] != "/" else base_url
+        self.token = GetToken(login_domain, auth0_client_id)
+        self.username = username
+        self.passwd = passwd
+        self._api_token = None
+
+    def authorize(self):
+        token_response = self.token.login(
+            username=self.username, password=self.passwd, realm='Username-Password-Authentication', grant_type='password'
+        )
+
+        self._api_token = f"{token_response['token_type']} {token_response['access_token']}"
+        self.headers.update({
+            "Authorization": self._api_token,
+        })
 
     def request(self, method, url, *args, **kwargs):
+        if self._api_token is None:
+            self.authorize()
+
         url = self.join_url(url)
+        response = super().request(method, url, *args, **kwargs)
+
+        if response.status_code == 401:
+            self.authorize()
```

Comment:
Probably here we should return response if status_code != 401 right ? Otherwise we make always two calls!